File: dotnet/Allmark/Render/ConsoleCommentRenderer.cs
namespace Allmark.Render;

using Allmark.Types;

public static class ConsoleCommentRenderer
{
    public static Renderer Create(string style, string reset)
    {
        return new Renderer
        {
            Name = "comment",
            Render = (node, state, first, last, decode) => Render(node, state, style, reset),
        };
    }

    public static Renderer Create()
    {
        return new Renderer
        {
            Name = "comment",
            Render = (node, state, first, last, decode) => Render(node, state, RenderToConsole.AnsiBlue, RenderToConsole.AnsiReset),
        };
    }

	public static void Render(MarkdownNode node, RendererState state, string style, string reset)
	{
		state.Output.Append($"{style}>>");
		RenderChildren.Execute(node, state);
		state.Output.Append($">>{reset}");
	}
}

File: swift/Sources/Allmark/RenderConsole/ConsoleCommentRenderer.swift
import Foundation


let consoleCommentRenderer = Renderer(
	name: "comment",
	render: renderConsoleComment
)


func renderConsoleComment(_ node: MarkdownNode, _ state: inout RendererState, _ first: Bool?, _ last: Bool?, _ decode: Bool?) {
	state.output += "<!--\(node.content)-->"
}
